# Exploring Ebay Car Sale Data
-------

The aim of this project is to clean the data and analyze the included used car listings. You'll also become familiar with some of the unique benefits jupyter notebook provides for pandas.

In [1]:
import pandas as pd
import numpy as np
autos = pd.read_csv('autos.csv',encoding="Latin-1")


In [2]:
autos.info()
autos.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

In [3]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [4]:
autos.columns = ['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'num_photos', 'postal_code',
       'last_seen']
autos.columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'num_photos', 'postal_code',
       'last_seen'],
      dtype='object')

In [5]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_ps   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  num_photos  postal_code  \
0     peugeot              nein  2016-03-26 00:00:00           0        79588   
1         bmw              nein  2016-04-04 00:00:00           0        71034   
2  volkswagen              nein  2016-03-26 00:00:00           0        35394   
3       smart              nein  2016-03-12 00:00:00           0        33729   
4        ford              nein  2016-04-01 00:00:00           0        39218   

             last_seen  
0  2016-04-06 06:45:54  
1  2016-04-06 14:45:08  
2  2016-04-06 20:15:37  
3  2016-03-15 03:16:28  
4  2016-04-01 14:38:50

In [6]:
autos.describe(include = "all")

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-03-10 15:36:24  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  registration_year  gearbox      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  num_photos   postal_code            last_seen  
count                 50000     50000.0  50000.000000                50000  
unique                   76         NaN           NaN                39481  
top     2016-04-03 00:00:00         NaN           NaN  2016-04-07 06:17:27  
freq                   1946         NaN           NaN                    8  
mean                    NaN         0.0  50813.627300                  NaN  
std                     NaN         0.0  25779.747957                  NaN  
min                     NaN         0.0   1067.000000                  NaN  
25%                     NaN         0.0  30451.000000                  NaN  
50%                     NaN         0.0  49577.000000                  NaN  
75%                     NaN         0.0  71540.000000                  NaN  
max                     NaN         0.0  99998.000000                  NaN

* num_photos,seller,offer_type have only one value so we should drop.
* power_ps need more investigation
* price and odometer needs to be cleaned

In [7]:
autos = autos.drop(["num_photos","seller","offer_type"],axis = 1)

In [8]:
autos["price"]= autos["price"].str.replace("$","").str.replace(",","").astype(int)
                            

In [9]:
autos["odometer"]= autos["odometer"].str.replace("km","").str.replace(",","").astype(int)


In [10]:
autos.rename({"odometer":"odometer_km"},axis = 1, inplace = True)

In [11]:
autos["odometer_km"].unique().shape

(13,)

In [12]:
autos["odometer_km"].describe()

count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [13]:
autos["odometer_km"].value_counts()

150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
70000      1230
60000      1164
50000      1027
5000        967
40000       819
30000       789
20000       784
10000       264
Name: odometer_km, dtype: int64

* Number of high mileage vehicles is higher than low mileage vehicles
* kilometer value is rounded so it must be entered through dropdown list/pre defined value set

In [14]:
autos["price"].unique().shape

(2357,)

In [15]:
autos["price"].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

* min price is $0 and max price is 100 million dollar which is too expensive

In [16]:
autos["price"].value_counts()

0         1421
500        781
1500       734
2500       643
1000       639
1200       639
600        531
800        498
3500       498
2000       460
999        434
750        433
900        420
650        419
850        410
700        395
4500       394
300        384
2200       382
950        379
1100       376
1300       371
3000       365
550        356
1800       355
5500       340
1250       335
350        335
1600       327
1999       322
          ... 
46200        1
29600        1
13480        1
21700        1
7373         1
3279         1
4286         1
188          1
17830        1
9130         1
910          1
238          1
2671         1
69900        1
151990       1
2479         1
4510         1
86500        1
47499        1
16998        1
27299        1
41850        1
4780         1
686          1
6495         1
20790        1
8970         1
846          1
2895         1
33980        1
Name: price, Length: 2357, dtype: int64

In [17]:
autos["price"].value_counts().sort_index(ascending=False).head(20)

99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
350000      1
345000      1
299000      1
295000      1
265000      1
259000      1
250000      1
220000      1
198000      1
197000      1
Name: price, dtype: int64

In [18]:
autos["price"].value_counts().sort_index(ascending=True).head(20)

0     1421
1      156
2        3
3        1
5        2
8        1
9        1
10       7
11       2
12       3
13       2
14       1
15       2
17       3
18       1
20       4
25       5
29       1
30       7
35       1
Name: price, dtype: int64

* There are 1421 cars with $0 price.

* Ebay is auction site so it is possible that $1 is starting bid so we will keep this items.

* We will remove the anything more than $350,000 as it seems unrealistic. 

In [19]:
autos = autos[autos["price"].between(1,350000)]

In [20]:
autos["price"].describe()

count     48565.000000
mean       5888.935591
std        9059.854754
min           1.000000
25%        1200.000000
50%        3000.000000
75%        7490.000000
max      350000.000000
Name: price, dtype: float64

In [21]:
autos[["date_crawled","ad_created","last_seen"]].head(5)

date_crawled           ad_created            last_seen
0  2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1  2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2  2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3  2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4  2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50

In [22]:
autos["date_crawled"].value_counts(normalize=True, dropna=False)

2016-03-08 10:40:35    0.000062
2016-03-14 20:50:02    0.000062
2016-03-23 19:38:20    0.000062
2016-03-25 19:57:10    0.000062
2016-03-30 19:48:02    0.000062
2016-03-22 09:51:06    0.000062
2016-03-19 17:36:18    0.000062
2016-03-16 21:50:53    0.000062
2016-03-12 16:06:22    0.000062
2016-03-29 23:42:13    0.000062
2016-03-23 18:39:34    0.000062
2016-04-04 16:40:33    0.000062
2016-03-11 22:38:16    0.000062
2016-04-02 15:49:30    0.000062
2016-03-05 16:57:05    0.000062
2016-03-21 20:37:19    0.000062
2016-03-09 11:54:38    0.000062
2016-04-02 11:37:04    0.000062
2016-03-27 19:53:12    0.000041
2016-03-27 14:58:21    0.000041
2016-03-15 13:36:19    0.000041
2016-04-03 20:54:10    0.000041
2016-03-27 16:39:15    0.000041
2016-03-15 18:57:09    0.000041
2016-03-24 20:58:12    0.000041
2016-03-17 10:49:44    0.000041
2016-03-10 19:57:04    0.000041
2016-03-31 19:41:18    0.000041
2016-03-29 22:06:23    0.000041
2016-03-10 15:36:24    0.000041
                         ...   
2016-03-

In [23]:
autos["date_crawled"].sort_index()

0        2016-03-26 17:47:46
1        2016-04-04 13:38:56
2        2016-03-26 18:57:24
3        2016-03-12 16:58:10
4        2016-04-01 14:38:50
5        2016-03-21 13:47:45
6        2016-03-20 17:55:21
7        2016-03-16 18:55:19
8        2016-03-22 16:51:34
9        2016-03-16 13:47:02
10       2016-03-15 01:41:36
11       2016-03-16 18:45:34
12       2016-03-31 19:48:22
13       2016-03-23 10:48:32
14       2016-03-23 11:50:46
15       2016-04-01 12:06:20
16       2016-03-16 14:59:02
17       2016-03-29 11:46:22
18       2016-03-26 19:57:44
19       2016-03-17 13:36:21
20       2016-03-05 19:57:31
21       2016-03-06 19:07:10
22       2016-03-28 20:50:54
23       2016-03-10 19:55:34
24       2016-04-03 11:57:02
25       2016-03-21 21:56:18
26       2016-04-03 22:46:28
28       2016-03-19 21:56:19
29       2016-04-02 12:45:44
30       2016-03-14 11:47:31
                ...         
49968    2016-04-01 17:49:15
49969    2016-03-17 18:49:02
49970    2016-03-21 22:47:37
49971    2016-

In [24]:
(autos["date_crawled"]
        .str[:10]
        .value_counts(normalize=True, dropna=False)
        .sort_values()
        )

2016-04-07    0.001400
2016-04-06    0.003171
2016-03-18    0.012911
2016-04-05    0.013096
2016-03-06    0.014043
2016-03-13    0.015670
2016-03-05    0.025327
2016-03-24    0.029342
2016-03-16    0.029610
2016-03-27    0.031092
2016-03-25    0.031607
2016-03-17    0.031628
2016-03-31    0.031834
2016-03-10    0.032184
2016-03-26    0.032204
2016-03-23    0.032225
2016-03-11    0.032575
2016-03-22    0.032987
2016-03-09    0.033090
2016-03-08    0.033296
2016-03-30    0.033687
2016-04-01    0.033687
2016-03-29    0.034099
2016-03-15    0.034284
2016-03-19    0.034778
2016-03-28    0.034860
2016-04-02    0.035478
2016-03-07    0.036014
2016-04-04    0.036487
2016-03-14    0.036549
2016-03-12    0.036920
2016-03-21    0.037373
2016-03-20    0.037887
2016-04-03    0.038608
Name: date_crawled, dtype: float64

The site was crawled almost daily in the month of March-April 2016 with uniform distribution

In [25]:
(autos["last_seen"]
        .str[:10]
        .value_counts(normalize=True, dropna=False)
        .sort_index()
        )


2016-03-05    0.001071
2016-03-06    0.004324
2016-03-07    0.005395
2016-03-08    0.007413
2016-03-09    0.009595
2016-03-10    0.010666
2016-03-11    0.012375
2016-03-12    0.023783
2016-03-13    0.008895
2016-03-14    0.012602
2016-03-15    0.015876
2016-03-16    0.016452
2016-03-17    0.028086
2016-03-18    0.007351
2016-03-19    0.015834
2016-03-20    0.020653
2016-03-21    0.020632
2016-03-22    0.021373
2016-03-23    0.018532
2016-03-24    0.019767
2016-03-25    0.019211
2016-03-26    0.016802
2016-03-27    0.015649
2016-03-28    0.020859
2016-03-29    0.022341
2016-03-30    0.024771
2016-03-31    0.023783
2016-04-01    0.022794
2016-04-02    0.024915
2016-04-03    0.025203
2016-04-04    0.024483
2016-04-05    0.124761
2016-04-06    0.221806
2016-04-07    0.131947
Name: last_seen, dtype: float64

* There is spike in the last three last_seen values.
* It is 6 to 10 times higher than other days.

In [26]:
autos["registration_year"].describe()

count    48565.000000
mean      2004.755421
std         88.643887
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

* min value is 1000 which is unreal
* max value is 9999 which is unreal too


In [27]:
# Removing incorrect data
#---------------------------------
autos = autos[autos["registration_year"].between(1900,2016)]

In [28]:
autos["registration_year"].describe()

count    46681.000000
mean      2002.910756
std          7.185103
min       1910.000000
25%       1999.000000
50%       2003.000000
75%       2008.000000
max       2016.000000
Name: registration_year, dtype: float64

In [29]:
autos["registration_year"].value_counts(normalize=True).head(10)

2000    0.067608
2005    0.062895
1999    0.062060
2004    0.057904
2003    0.057818
2006    0.057197
2001    0.056468
2002    0.053255
1998    0.050620
2007    0.048778
Name: registration_year, dtype: float64

In [30]:
#Exploring Brand coloumn
autos["brand"].unique().shape[0]

40

In [31]:
autos["brand"].value_counts(normalize = True)

volkswagen        0.211264
bmw               0.110045
opel              0.107581
mercedes_benz     0.096463
audi              0.086566
ford              0.069900
renault           0.047150
peugeot           0.029841
fiat              0.025642
seat              0.018273
skoda             0.016409
nissan            0.015274
mazda             0.015188
smart             0.014160
citroen           0.014010
toyota            0.012703
hyundai           0.010025
sonstige_autos    0.009811
volvo             0.009147
mini              0.008762
mitsubishi        0.008226
honda             0.007840
kia               0.007069
alfa_romeo        0.006641
porsche           0.006127
suzuki            0.005934
chevrolet         0.005698
chrysler          0.003513
dacia             0.002635
daihatsu          0.002506
jeep              0.002271
subaru            0.002142
land_rover        0.002099
saab              0.001649
jaguar            0.001564
daewoo            0.001500
trabant           0.001392
r

In [32]:
#top brands
brand_count = autos["brand"].value_counts(normalize = True)
common_brand = brand_count[brand_count > 0.05].index

In [33]:
print(common_brand) # top common brands

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford'], dtype='object')


In [34]:
# dictionary to store brands and mean prices
brand_mean_prices = {}

for b in common_brand:
    brand_only = autos[autos["brand"]==b]
    mean_price = brand_only["price"].mean()
    brand_mean_prices[b] = int(mean_price)
    
brand_mean_prices

{'audi': 9336,
 'bmw': 8332,
 'ford': 3749,
 'mercedes_benz': 8628,
 'opel': 2975,
 'volkswagen': 5402}

Here, we can see that Audi, Mercedes Benz and BMW are more expensive.


In [35]:
brand_mean_mileage={}

for b in common_brand:
    brand_only = autos[autos["brand"]==b]
    mean_mileage = brand_only["odometer_km"].mean()
    brand_mean_mileage[b] = int(mean_mileage)
    
brand_mean_mileage

{'audi': 129157,
 'bmw': 132572,
 'ford': 124266,
 'mercedes_benz': 130788,
 'opel': 129310,
 'volkswagen': 128707}

In [36]:
#convert dictionaries into series to combine both results
mean_mileage = pd.Series(brand_mean_mileage).sort_values(ascending=False)
mean_prices = pd.Series(brand_mean_prices).sort_values(ascending=False)

In [37]:
df_mileage = pd.DataFrame(mean_mileage,columns = ['mean_mileage'])
df_mileage

mean_mileage
bmw                  132572
mercedes_benz        130788
opel                 129310
audi                 129157
volkswagen           128707
ford                 124266

In [38]:
brand_info= pd.DataFrame(mean_prices,columns=['mean_price'])
brand_info

mean_price
audi                 9336
mercedes_benz        8628
bmw                  8332
volkswagen           5402
ford                 3749
opel                 2975

In [39]:
brand_info['mean_mileage'] = df_mileage
brand_info

mean_price  mean_mileage
audi                 9336        129157
mercedes_benz        8628        130788
bmw                  8332        132572
volkswagen           5402        128707
ford                 3749        124266
opel                 2975        129310

In [40]:
autos["gearbox"].head(5)

0      manuell
1    automatik
2      manuell
3    automatik
4      manuell
Name: gearbox, dtype: object

In [41]:
mapping_dict = {
    "automatik":"Automatic",
    "manuell":"Manual"}

autos["gearbox"]= autos["gearbox"].map(mapping_dict)

In [42]:
autos["gearbox"].head(5)

0       Manual
1    Automatic
2       Manual
3    Automatic
4       Manual
Name: gearbox, dtype: object

In [43]:
# # mapping_dict = {
#     "automatik":"Automatic",
#     "manuell":"Manual"}

# autos["gearbox"]= autos["gearbox"].map(mapping_dict)
autos.describe(include = "all")

date_crawled                 name          price abtest  \
count                 46681                46681   46681.000000  46681   
unique                45124                35812            NaN      2   
top     2016-04-02 11:37:04  Volkswagen_Golf_1.4            NaN   test   
freq                      3                   75            NaN  24062   
mean                    NaN                  NaN    5977.716801    NaN   
std                     NaN                  NaN    9177.909479    NaN   
min                     NaN                  NaN       1.000000    NaN   
25%                     NaN                  NaN    1250.000000    NaN   
50%                     NaN                  NaN    3100.000000    NaN   
75%                     NaN                  NaN    7500.000000    NaN   
max                     NaN                  NaN  350000.000000    NaN   

       vehicle_type  registration_year gearbox      power_ps  model  \
count         43977       46681.000000   44571  46681.000000  44488   
unique            8                NaN       2           NaN    244   
top       limousine                NaN  Manual           NaN   golf   
freq          12598                NaN   34715           NaN   3707   
mean            NaN        2002.910756     NaN    117.892933    NaN   
std             NaN           7.185103     NaN    184.922911    NaN   
min             NaN        1910.000000     NaN      0.000000    NaN   
25%             NaN        1999.000000     NaN     75.000000    NaN   
50%             NaN        2003.000000     NaN    109.000000    NaN   
75%             NaN        2008.000000     NaN    150.000000    NaN   
max             NaN        2016.000000     NaN  17700.000000    NaN   

          odometer_km  registration_month fuel_type       brand  \
count    46681.000000        46681.000000     43363       46681   
unique            NaN                 NaN         7          40   
top               NaN                 NaN    benzin  volkswagen   
freq              NaN                 NaN     28540        9862   
mean    125586.855466            5.827125       NaN         NaN   
std      39852.528628            3.670300       NaN         NaN   
min       5000.000000            0.000000       NaN         NaN   
25%     100000.000000            3.000000       NaN         NaN   
50%     150000.000000            6.000000       NaN         NaN   
75%     150000.000000            9.000000       NaN         NaN   
max     150000.000000           12.000000       NaN         NaN   

       unrepaired_damage           ad_created   postal_code  \
count              38374                46681  46681.000000   
unique                 2                   74           NaN   
top                 nein  2016-04-03 00:00:00           NaN   
freq               33834                 1821           NaN   
mean                 NaN                  NaN  51097.434181   
std                  NaN                  NaN  25755.387192   
min                  NaN                  NaN   1067.000000   
25%                  NaN                  NaN  30827.000000   
50%                  NaN                  NaN  49828.000000   
75%                  NaN                  NaN  71732.000000   
max                  NaN                  NaN  99998.000000   

                  last_seen  
count                 46681  
unique                37146  
top     2016-04-07 06:17:27  
freq                      8  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN

In [46]:
mapping_dict = {
    "no":"undamaged",
    "yes":"damaged"}

autos["unrepaired_damage"]= autos["unrepaired_damage"].map(mapping_dict)
autos["unrepaired_damage"].head(50)

0     undamaged
1     undamaged
2     undamaged
3     undamaged
4     undamaged
5           NaN
6           NaN
7     undamaged
8     undamaged
9     undamaged
11          NaN
12    undamaged
13    undamaged
14          NaN
15    undamaged
16    undamaged
17    undamaged
18          NaN
19    undamaged
20          NaN
21    undamaged
22    undamaged
23    undamaged
24    undamaged
25      damaged
26    undamaged
28      damaged
29    undamaged
30      damaged
31    undamaged
32          NaN
33    undamaged
34          NaN
35    undamaged
36    undamaged
37    undamaged
38    undamaged
39    undamaged
40    undamaged
41    undamaged
42          NaN
43    undamaged
44    undamaged
45    undamaged
46          NaN
47    undamaged
48    undamaged
49    undamaged
50    undamaged
51      damaged
Name: unrepaired_damage, dtype: object